### PR TITLE
MUSL target, dependencies compiler checker

### PR DESCRIPTION
**Pull request by GANitak**
Here i don't know if that the way to pull in the target checks, i just looked at how future highway did it with the buf one and added the Filesystem Layout Dependencies for musl 
* But line 179 is i *suppose* is where this checker should be added

### CODE REVIEW

**Review by future-highway — CHANGES_REQUESTED** [review]
The buf check is different, you should ignore that for this.

The goal is to put a check on the musl target in the main Makefile. We have another Makefile in the project (under the ebpf directory) that does a check using grep.

Feel free to ping in discord if you want more specifics.

File: Makefile [hunk 1]
```diff
@@ -176,7 +177,7 @@ PROGS = aer auraed auraescript
 define AURAE_template =
 .PHONY: $(1)
 $(1): musl $(GEN_RS) $(GEN_TS) $(1)-lint $(1)-debug ## Lint and install $(1) (for use during development)
-
+
```
Comment (future-highway) [inline Makefile:180]:
```suggestion

```

File: Makefile [hunk 2]
```diff
@@ -116,7 +116,8 @@ config: certs ## Set up default config
 
 .PHONY: musl
 musl: ## Add target for musl
-	rustup target add $(uname_m)-unknown-linux-musl
+	rustup target list | grep -qc $(uname_m)-unknown-linux-musl || \
```
Comment (future-highway) [inline Makefile:119]: This line is really close. But:

1. grep takes a string for the pattern input (run `grep --help`)
2. when you run `rustup target list` you'll notice it lists all targets (not just those that are installed). The ones that are installed are specifically indicated

File: Makefile [hunk 3]
```diff
@@ -116,7 +116,8 @@ config: certs ## Set up default config
...
+	rustup target install $(uname_m)-unknow-linux-muslknown-linux-musl
```
Comment (future-highway) [inline Makefile:120]: You can run `rustup target --help` to see the available subcommands. "install" is not an option.

**Review by future-highway — CHANGES_REQUESTED** [review]
Much closer

File: Makefile (hunk 3 above)
Comment (GANitak) [reply Makefile:120]: so it's rustup target add

File: Makefile [hunk 4]
```diff
@@ -116,7 +116,8 @@ config: certs ## Set up default config
 
 .PHONY: musl
 musl: ## Add target for musl
-	rustup target add $(uname_m)-unknown-linux-musl
+	rustup target list | grep -qc '$(uname_m)-unknown-linux-musl' || \
+	rustup target add $(uname_m)-unknow-linux-muslknown-linux-musl
```
Comment (future-highway) [inline Makefile:120]: You're almost done. Just looks like the name of the target here got changed accidentally.

**Review by future-highway — CHANGES_REQUESTED** [review]

File: Makefile [hunk 5]
```diff
@@ -116,7 +116,8 @@ config: certs ## Set up default config
 
 .PHONY: musl
 musl: ## Add target for musl
-	rustup target add $(uname_m)-unknown-linux-musl
+	rustup target list | grep -qc '$(uname_m)-unknown-linux-musl' || \
```
Comment (future-highway) [inline Makefile:119]: The fix to making it a string was correct, but the string itself is a little off. Installed targets (what we want grep to look for) end in " (installed)"

**Review by future-highway — CHANGES_REQUESTED** [review]

File: Makefile [hunk 6]
```diff
@@ -116,7 +116,8 @@ config: certs ## Set up default config
 
 .PHONY: musl
 musl: ## Add target for musl
-	rustup target add $(uname_m)-unknown-linux-musl
+	rustup target list | grep -qc '$(uname_m)-unknown-linux-musl (installed)' || \
+	rustup target add $(uname_m)-unknow-linux-musl
```
Comment (future-highway) [inline Makefile:120]:
```suggestion
	rustup target add $(uname_m)-unknown-linux-musl
```

**Review by future-highway — CHANGES_REQUESTED** [review]

**Review by future-highway — APPROVED** [review]
Got it!

**Pull request merged**